### ISSUE TITLE
Training issues on custom data

### ISSUE THREAD

**Issue by crazyggg1** [naver/dust3r#221]
Thanks for your great works! I'm trying to finetune DUST3R on my own dataset, and I froze some of the parameters. However, the validation loss seems unstable and doesn't decrease steadily during training. Could you provide some insight suggestion on these issues? 

![Image](https://github.com/user-attachments/assets/2698d9e5-7903-4915-90b9-50470c42169b)

![Image](https://github.com/user-attachments/assets/191f2e52-28d8-4457-a1b8-3b96040ff10a)

**Comment by ZhaoLl-code** [comment]
Hello, it seems you're also working on training the DUSt3R model. I've encountered some issues during my training process and was hoping we could discuss them together. For example, is it necessary to perform point cloud registration on the dataset before using it for training? Additionally, are you using the loss function from the original source code or one you've written yourself?

**Comment by Henry-Bi** [comment]
I am trying to finetune dust3r to my own datasets. The loss functions are adopted from the original implementation. If you want to finetune dust3r, you should prepara image pairs as mentinoned in the readme.

**Comment by ZhaoLl-code** [comment]
Thank you for your reply. I started training with the MV-DUSt3R model. However, I found that although the loss functions of DUSt3R and MV-DUSt3R are similar in the paper, there is a significant difference between them in the source code. Therefore, I tried to rewrite a loss function based on the content in the paper to replace the one in the MV-DUSt3R source code. If possible, I would like to share my version with you and discuss it together.

> I am trying to finetune dust3r to my own datasets. The loss functions are adopted from the original implementation. If you want to finetune dust3r, you should prepara image pairs as mentinoned in the readme.

**Comment by ZhaoLl-code** [comment]
> > Thank you for your reply. I started training with the MV-DUSt3R model. However, I found that although the loss functions of DUSt3R and MV-DUSt3R are similar in the paper, there is a significant difference between them in the source code. Therefore, I tried to rewrite a loss function based on the content in the paper to replace the one in the MV-DUSt3R source code. If possible, I would like to share my version with you and discuss it together.
> > > I am trying to finetune dust3r to my own datasets. The loss functions are adopted from the original implementation. If you want to finetune dust3r, you should prepara image pairs as mentinoned in the readme.
> 
> Sounds great! I'm excited to dive deeper into this with you. Would you mind sharing your preferred way of staying in touch? Email, WeChat, QQ, or any other platform you use would be fine with me.

QQ all right？1367918192

**Comment by starInUniverse** [comment]
Hello, I'm trying to train DUST3R on my own dataset, too. But the result in predicting model and global-alignment will be strange. Could you talk about how you preprocess your own dataset, please? I just try the colmap and the agisoft. But it doesn't seem work in the camera poses.

**Comment by Henry-Bi** [comment]
> Hello, I'm trying to train DUST3R on my own dataset, too. But the result in predicting model and global-alignment will be strange. Could you talk about how you preprocess your own dataset, please? I just try the colmap and the agisoft. But it doesn't seem work in the camera poses.

You can evaluate the paired point clouds using the following script in the dust3r repository:
[waymo.py#L69C1-L93C19](https://github.com/naver/dust3r/blob/ad9d1a0421b19d7be72801f6e2e8bfcf79dc3c68/dust3r/datasets/waymo.py#L69C1-L93C19)

If the point clouds from different views appear misaligned, you should consider whether the coordinate system transformation is consistent. Here's how I apply the transformation:
`R, T = camera_pose[:3, :3], camera_pose[:3, 3:]
R = np.stack([R[:, 0], -R[:, 1], -R[:, 2]], axis=1)
camera_pose = np.concatenate([R, T], axis=1)
camera_pose = np.float32(np.vstack((camera_pose, np.array([[0, 0, 0, 1]]))))
`

**Comment by crazyggg1** [comment]
> > > Hello, I'm trying to train DUST3R on my own dataset, too. But the result in predicting model and global-alignment will be strange. Could you talk about how you preprocess your own dataset, please? I just try the colmap and the agisoft. But it doesn't seem work in the camera poses.
> > 
> > 
> > You can evaluate the paired point clouds using the following script in the dust3r repository: [waymo.py#L69C1-L93C19](https://github.com/naver/dust3r/blob/ad9d1a0421b19d7be72801f6e2e8bfcf79dc3c68/dust3r/datasets/waymo.py#L69C1-L93C19)
> > If the point clouds from different views appear misaligned, you should consider whether the coordinate system transformation is consistent. Here's how I apply the transformation: `R, T = camera_pose[:3, :3], camera_pose[:3, 3:] R = np.stack([R[:, 0], -R[:, 1], -R[:, 2]], axis=1) camera_pose = np.concatenate([R, T], axis=1) camera_pose = np.float32(np.vstack((camera_pose, np.array([[0, 0, 0, 1]])))) `
> 
> Sorry for the late reply. Thank you so much for your suggestion! I have tried evaluating the paired point clouds on my own datasets, and they aligned well.
> 
> ![Image](https://github.com/user-attachments/assets/7195b596-612d-4f84-9faa-dfc3ac08ffc5)
> 
> But using my own datasets to train, I got the misaligned prediction like this:
> 
> ![Image](https://github.com/user-attachments/assets/d30f523a-7f4f-415c-9bcd-e87331b81276)
> 
> Do you have any idea about this problem? Thank you again!

I'm experiencing similar misalignment issues when training DUST3R on my own datasets. Based on my analysis, this might be caused by several factors: finetune strategy and domain shift between DUST3R data and my owns. I'm still working on resolving this issue completely. Would love to hear your thoughts and any additional insights from the community!

**Comment by starInUniverse** [comment]
> > > > Hello, I'm trying to train DUST3R on my own dataset, too. But the result in predicting model and global-alignment will be strange. Could you talk about how you preprocess your own dataset, please? I just try the colmap and the agisoft. But it doesn't seem work in the camera poses.
> > > 
> > > 
> > > You can evaluate the paired point clouds using the following script in the dust3r repository: [waymo.py#L69C1-L93C19](https://github.com/naver/dust3r/blob/ad9d1a0421b19d7be72801f6e2e8bfcf79dc3c68/dust3r/datasets/waymo.py#L69C1-L93C19)
> > > If the point clouds from different views appear misaligned, you should consider whether the coordinate system transformation is consistent. Here's how I apply the transformation: `R, T = camera_pose[:3, :3], camera_pose[:3, 3:] R = np.stack([R[:, 0], -R[:, 1], -R[:, 2]], axis=1) camera_pose = np.concatenate([R, T], axis=1) camera_pose = np.float32(np.vstack((camera_pose, np.array([[0, 0, 0, 1]])))) `
> > 
> > 
> > Sorry for the late reply. Thank you so much for your suggestion! I have tried evaluating the paired point clouds on my own datasets, and they aligned well.
> > ![Image](https://github.com/user-attachments/assets/7195b596-612d-4f84-9faa-dfc3ac08ffc5)
> > But using my own datasets to train, I got the misaligned prediction like this:
> > ![Image](https://github.com/user-attachments/assets/d30f523a-7f4f-415c-9bcd-e87331b81276)
> > Do you have any idea about this problem? Thank you again!
> 
> I'm experiencing similar misalignment issues when training DUST3R on my own datasets. Based on my analysis, this might be caused by several factors: finetune strategy and domain shift between DUST3R data and my owns. I'm still working on resolving this issue completely. Would love to hear your thoughts and any additional insights from the community!

I think it's my camera poses may have some differences with the dust3r, as you mentioned, the domain shift. I am still trying to figure it out, and I use colmap to get the world2cam, and then I transfer it to cam2world. The dust3r uses the opencv-poses, and its coordinate system is RDF, the same with the colmap. Then it really puzzles me. And I also doubt it is the global alignment that causes our problem. Have you make any further understanding?
